Content.Shared/Doors/Systems/SharedAirlockSystem.cs

```diff
@@ -123,11 +125,23 @@ public void UpdateEmergencyLightStatus(EntityUid uid, AirlockComponent component
         Appearance.SetData(uid, DoorVisuals.EmergencyLights, component.EmergencyAccess);
     }
 
-    public void ToggleEmergencyAccess(EntityUid uid, AirlockComponent component)
+    public void SetEmergencyAccess(Entity<AirlockComponent> ent, bool value, EntityUid? user = null, bool predicted = false)
     {
-        component.EmergencyAccess = !component.EmergencyAccess;
-        Dirty(uid, component); // This only runs on the server apparently so we need this.
-        UpdateEmergencyLightStatus(uid, component);
+        if(!ent.Comp.Powered)
+            return;
+
+        if (ent.Comp.EmergencyAccess == value)
+            return;
+
+        ent.Comp.EmergencyAccess = value;
+        Dirty(ent, ent.Comp); // This only runs on the server apparently so we need this.
+        UpdateEmergencyLightStatus(ent, ent.Comp);
+		
+        var sound = ent.Comp.EmergencyAccess ? ent.Comp.EmergencyOnSound : ent.Comp.EmergencyOffSound;
+        if (predicted)
+            Audio.PlayPredicted(sound, ent, user: user);
+        else
+            Audio.PlayPvs(sound, ent);
     }
```

Comment:
Do any of the callers even need this? This method is new...

Reply:
AdminVerbSystem uses non-predicted version. is that wrong?

Reply:
That's fine for now.